data/npc/scripts/bank.lua

```diff
@@ -0,0 +1,339 @@
+			end
+			npcHandler:say("So you would like to transfer " .. count[cid] .. " gold to " .. transfer[cid] .. "?", cid)
+			npcHandler.topic[cid] = 13
+		else
+			npcHandler:say("This player does not exist.", cid)
+			npcHandler.topic[cid] = 0
+		end
+	elseif npcHandler.topic[cid] == 13 then
+		if msgcontains(msg, "yes") then
+			if not player:transferMoneyTo(transfer[cid], count[cid]) then
+				npcHandler:say("You cannot transfer money to this account.", cid)
+			else
+				npcHandler:say("Very well. You have transfered " .. count[cid] .. " gold to " .. transfer[cid] ..".", cid)
+				transfer[cid] = nil
+			end
+		elseif msgcontains(msg, "no") then
+			npcHandler:say("Alright, is there something else I can do for you?", cid)
+		end
+		npcHandler.topic[cid] = 0
+	elseif msgcontains(msg, "change gold") then
+		npcHandler:say("How many platinum coins would you like to get?", cid)
+		npcHandler.topic[cid] = 14
+	elseif npcHandler.topic[cid] == 14 then
+		if getMoneyCount(msg) < 1 then
+			npcHandler:say("Sorry, you do not have enough gold coins.", cid)
+			npcHandler.topic[cid] = 0
+		else
+			count[cid] = getMoneyCount(msg)
+			npcHandler:say("So you would like me to change " .. count[cid] * 100 .. " of your gold coins into " .. count[cid] .. " platinum coins?", cid)
+			npcHandler.topic[cid] = 15
+		end
+	elseif npcHandler.topic[cid] == 15 then
+		if msgcontains(msg, "yes") then
+			if player:removeItem(ITEM_GOLD_COIN, count[cid] * 100) then
+				player:addItem(ITEM_PLATINUM_COIN, count[cid])
+				npcHandler:say("Here you are.", cid)
+			else
+				npcHandler:say("Sorry, you do not have enough gold coins.", cid)
+			end
+		else
+			npcHandler:say("Well, can I help you with something else?", cid)
+		end
+		npcHandler.topic[cid] = 0
+	elseif msgcontains(msg, "change platinum") then
+		npcHandler:say("Would you like to change your platinum coins into gold or crystal?", cid)
+		npcHandler.topic[cid] = 16
+	elseif npcHandler.topic[cid] == 16 then
+		if msgcontains(msg, "gold") then
+			npcHandler:say("How many platinum coins would you like to change into gold?", cid)
+			npcHandler.topic[cid] = 17
+		elseif msgcontains(msg, "crystal") then
+			npcHandler:say("How many crystal coins would you like to get?", cid)
+			npcHandler.topic[cid] = 19
+		else
+			npcHandler:say("Well, can I help you with something else?", cid)
+			npcHandler.topic[cid] = 0
+		end
+	elseif npcHandler.topic[cid] == 17 then
+		if getMoneyCount(msg) < 1 then
+			npcHandler:say("Sorry, you do not have enough platinum coins.", cid)
+			npcHandler.topic[cid] = 0
+		else
+			count[cid] = getMoneyCount(msg)
+			npcHandler:say("So you would like me to change " .. count[cid] .. " of your platinum coins into " .. count[cid] * 100 .. " gold coins for you?", cid)
+			npcHandler.topic[cid] = 18
+		end
+	elseif npcHandler.topic[cid] == 18 then
+		if msgcontains(msg, "yes") then
+			if player:removeItem(ITEM_PLATINUM_COIN, count[cid]) then
+				player:addItem(ITEM_GOLD_COIN, count[cid] * 100)
```

Comment:
player capacity isn't being checked when changing platinum to gold and crystal to platinum. I saw earlier in the script that capacity was checked before money is withdrawn. I don't recall what cipbia does in this case. So i'm not sure if this is a bug or silly cipbia behavior is being mirrored.

hmm.. I can't recall the exact situation but I do remember using the banker in cipbia and dropping gold on the floor. Maybe this happens there is not enough open container spaces?

Reply:
indeed, player capacity should be checked here as well, I just noticed that the getFreeCapacity is being called only once in this script on the withdraw function, but not in any of the change type function, I'll address this ASAP

Reply:
added the check, please test it if possible